### PR TITLE
Add bar repeat feature as a plugin option

### PR DESCRIPTION
**Pull request by unclechu**
I just have written this for myself in order to be able to change wideness of the bar more easily, potentially dynamically, and decided to share this as a usage example.

My config of the bar: https://github.com/unclechu/neovimrc/blob/01bbed8d85c86a44140a44eca4baf508e4a5b0c5/plugins-configs.vim#L53-L74

_P.S. I found this plug-in repo from Mastering Vim Quickly e-mail newsletter._

**UPD:**

Added an option `g:line_no_indicator_bar_repeats` with a number of repeats.
Added usage examples to `README.md`.

### CODE REVIEW

**Comment by drzel** [comment]
Cool any chance of integrating this into the plugin and adding an option?

**Comment by unclechu** [comment]
@drzel Sure, it seems it should be pretty simple and straightforward. I'll let you know when I come up with something. Marking this as **WIP** for now.

**Comment by unclechu** [comment]
@drzel It's done, examples in `README.md` are also updated.
Added an option `g:line_no_indicator_bar_repeats` with a number of repeats.

**Comment by unclechu** [comment]
@drzel Are you going to test it later and merge?

File: README.md [hunk 1]
```diff
@@ -89,25 +89,31 @@ Some alternatives:
...
-  \ '█▎ ', '█▍ ', '█▌ ', '█▋ ', '█▊ ',
-  \ '█▉ ', '██ ', '██▏', '██▎', '██▍',
-  \ '██▌', '██▋', '██▊', '██▉', '███'
+let g:line_no_indicator_chars =
+  \ [ '   ', '▏  ', '▎  ', '▍  ', '▌  '
+  \ , '▋  ', '▊  ', '▉  ', '█  ', '█▏ '
+  \ , '█▎ ', '█▍ ', '█▌ ', '█▋ ', '█▊ '
+  \ , '█▉ ', '██ ', '██▏', '██▎', '██▍'
+  \ , '██▌', '██▋', '██▊', '██▉', '███'
   \ ]
```
Comment (drzel) [inline README.md:106]: Is there a style guide or some other resource for why one style is preferable?

**Comment by drzel** [comment]
When I install this branch I get:

```
Error detected while processing function LineNoIndicator:
line    9:
E15: Invalid expression: *= g:line_no_indicator_bar_repeats
Error detected while processing function LineNoIndicator:
line    9:
E488: Trailing characters
```

**Comment by unclechu** [comment]
@drzel Hm... Okay, I'll fix it, by the way, what version of Neovim/Vim are you using?

**Comment by unclechu** [comment]
@drzel I got rid of those operators, now it has to work.

File: plugin/line-no-indicator.vim [hunk 2]
```diff
@@ -18,19 +18,32 @@ if !exists('g:line_no_indicator_chars')
   end
 end
 
+if !exists('g:line_no_indicator_bar_repeats')
+  let g:line_no_indicator_bar_repeats = 1
+en
```
Comment (drzel) [inline plugin/line-no-indicator.vim:23]: If possible can we use full-length commands e.g. `endif`?

**Comment by unclechu** [comment]
@drzel bump after a week

**Comment by unclechu** [comment]
@drzel ping

**Comment by drzel** [comment]
Hey @unclechu I left a few comments on the PR that you might have missed. I don't think this feature really adds much from a user perspective, so while I'm happy to add it, I don't want to make the code more complicated. I find it hard to follow at the moment - probably my fault. Could you use more explicit local variable names to to make it clear what's happening at each step?

Thanks

**Comment by unclechu** [comment]
@drzel Could you please give me the links to those comments, for some reason I can't find them, anywhere here in the PR.

I'll add some comments for the code soon.
And I'll try to find better name for `l:single_len` variable but I don't know what's wrong with `l:result`, isn't it as explicit as possible? I've added only these two variables (except global one for new option).

> I don't think this feature really adds much from a user perspective

I don't know, but for me it's really useful, since I had to implement this manually in my config. And the only reason I've created this PR is that I considered that user's config is definitely not the right place for this chunk of code.

**Comment by unclechu** [comment]
@drzel Okay, I've added many comments trying to explain what every variable's value is. Also renamed that `l:single_len` variable.

Is it okay now?

**Comment by unclechu** [comment]
@drzel ping?

**Review by drzel — COMMENTED** [review]
Hey it seems a little bit complicated to me. I feel like there is a simpler way to do it - I'd like to review it in more detail when I get time. Thanks for sticking with it though.

File: README.md (hunk 1 above)
Comment (unclechu) [reply README.md:106]: @drzel It's just brought this style from Haskell, it's personal preference, so you'd never miss a comma at the end of a line.

And I use one-liner list if it fits one line of 80 chars.

I'll change it to your initial style, it doesn't matter much.

File: plugin/line-no-indicator.vim (hunk 2 above)
Comment (unclechu) [reply plugin/line-no-indicator.vim:23]: @drzel NP, I'll change it.

**Comment by unclechu** [comment]
I decided to maintain my own fork. I don't think there is simpler way to do it. Since this MR is going to be abandoned forever I'm closing it.
